### PR TITLE
Add logging in segment upload logic to make refresh scenarios traceable in logs

### PR DESCRIPTION
**Pull request by npawar**

### CODE REVIEW

**Comment by npawar** [comment]
@mcvsubbu From our discussion this morning, adding logs to the segment upload code to be able to trace refresh cases. Please let me know if you think more is required

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java [hunk 1]
```diff
@@ -517,14 +522,18 @@ private void uploadSegment(File indexDir, SegmentMetadata segmentMetadata, File
 
       // Update ZK metadata and refresh the segment if necessary
       long newCrc = Long.valueOf(segmentMetadata.getCrc());
+      LOGGER.info("CRC for segment {}, existing: {} new: {}", segmentName, existingCrc, newCrc);
       if (newCrc == existingCrc) {
-        // New segment is the same as the existing one, only update ZK metadata without refresh the segment
+        LOGGER.info("New segment {} is same as existing one, updating ZK metadata without refreshing the segment",
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java]: might want to change the statement to say that new segment crc is same as existing segment crc

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java [hunk 2]
```diff
@@ -517,14 +522,18 @@ private void uploadSegment(File indexDir, SegmentMetadata segmentMetadata, File
...
+            segmentName);
         if (!_pinotHelixResourceManager.updateZkMetadata(existingSegmentZKMetadata)) {
           throw new RuntimeException(
               "Failed to update ZK metadata for segment: " + segmentName + " of table: " + offlineTableName);
         }
       } else {
         // New segment is different with the existing one, update ZK metadata and refresh the segment
+        LOGGER.info("New segment {} is different than the existing one, updating ZK metadata and refresh segment",
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java]: same as above

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java [hunk 3]
```diff
@@ -407,6 +407,9 @@ private void uploadSegment(File indexDir, SegmentMetadata segmentMetadata, File
     TableConfig offlineTableConfig =
         ZKMetadataProvider.getOfflineTableConfig(_pinotHelixResourceManager.getPropertyStore(), offlineTableName);
 
+    LOGGER.info("Starting upload for segment {}, tempTarredSegmentFile {}", segmentName,
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java]: may want to add a comment here (after verification) that the time difference between this log and the one that logs the headers gives us the time taken to untar the segment. Useful for debugging if storage is slow.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java (hunk 2 above)
Comment (npawar) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java]: done

**Pull request merged**